test/client_test.tsx

```diff
@@ -30,6 +25,49 @@ import {
 } from "../src/types/ws"
 import { virtualize, concretize, STYLE_TAG_ID } from "../src/helpers"
 
+test("window events are dispatched correctly", async () => {
+  const events: PurviewWebsocketEvent[] = []
+  const eventHandler = (e: CustomEvent<PurviewWebsocketEvent>) => {
+    events.push(e.detail)
+  }
+  window.addEventListener("purview", eventHandler as EventListener)
```

Comment:
This should not be necessary. If we are loading purview, it should augment the available window events automatically with the proper type via declaration merging, e.g.:

```ts
declare global {
  interface WindowEventMap {
    "purview": CustomEvent<PurviewWebsocketEvent>;
  } 
}
```

This ensures that an event handler for `"purview"` added via `window.addEventListener` has the correct type inferred without a type assertion.